tests/loop_rolling/test_loop_rolling.py

```diff
@@ -0,0 +1,198 @@
+import argparse
+
+import numpy as np
+
+import onnx
+import os
+
+from onnxscript import ir
+from onnxscript import rewriter
+
+#.rewriter import PatternRewriteRule, RewriteRuleSet, rewrite
+import onnxruntime as onnxrt
+
+from onnxscript.rewriter import rewrite, RewriteRuleSet
+
+from onnxscript.rewriter.pattern_builder_jsm import build_loop_replace_pattern
+from onnxscript.rewriter.pattern_builder_jsm import normalize_io_for_loop_rolling
+from onnxscript.rewriter.pattern_builder_jsm import LoopBodyTemplate
+
+from onnx import shape_inference
+
+def remove_existing_data_file(filename):
+    if os.path.exists(filename):
+        print(f"Removing existing data file: {filename}")
+        os.remove(filename)
+
+def open_ir(filename):
+    print('loading onnx')
+    f = onnx.load(filename)
+    print('deserializing')
+    return ir.serde.deserialize_model(f)
+
+# Handle Parser Stuff
+parser = argparse.ArgumentParser(description="A simple argparse example")
+parser.add_argument("filename", type=str, help="The name of the input onnx file")
+parser.add_argument("patternfilename", type=str, help="The name of the input onnx file that is a single layer")
+args = parser.parse_args()
+
+def ort_make_rand_io(filename):
+
+    session = onnxrt.InferenceSession(filename)
+
+    input_name  = session.get_inputs()[0].name
+    input_type  = session.get_inputs()[0].type
+    input_shape = session.get_inputs()[0].shape
+
+    if input_type == 'tensor(int64)':
+        np_type = np.int64
+    elif input_type == 'tensor(float)':
+        np_type = np.float32
+    else:
+        raise Exception("unsupported type {input_type}")
+
+    input_data  = np.random.random(input_shape).astype(np_type)
+
+    return ({input_name: input_data}, session.get_outputs())
+
+
+def ort_run_graph(filename, input_dict, output_name):
+    sess_options = onnxrt.SessionOptions()
+    sess_options.log_severity_level = 0
+    sess_options.graph_optimization_level = onnxrt.GraphOptimizationLevel.ORT_ENABLE_BASIC
+
+    return onnxrt.InferenceSession(filename, sess_options=sess_options).run([output_name], input_dict)
+
+input_dict, outputs = ort_make_rand_io(args.filename)
+golden_results      = ort_run_graph(args.filename, input_dict, outputs[0].name)
+
+
+LoopBody = LoopBodyTemplate(args.patternfilename)
```

Comment:
```suggestion
loop_body = LoopBodyTemplate(args.patternfilename)
```

note: always use snake_case for variable names